strings/string_is_valid_number.py

```diff
@@ -0,0 +1,134 @@
+"""
+Solution By: Reniz Shah
+Topic: Deterministic Finite Automaton (DFA)
+Given a string s, return whether s is a valid number or not
+Input: s = -90E3
+Output: True
+Leetcode link: https://leetcode.com/problems/valid-number/description/
+"""
+
+
+def classify_char(char):
```

Comment:
Please read CONTRIBUTING.md
All functions in this repo are required to have type hints.
```suggestion
def classify_char(char: str) -> str | None:
```